### ISSUE TITLE
'<' not supported between instances of 'NoneType' and 'str'

### ISSUE THREAD

**Issue by hkaancaliskan** [archlinux/archinstall#783]
I was trying to dual boot with windows.
/dev/nvme0n1p1 -> EFI -> 1GB
/dev/nvme0n1p2 -> MSR
/dev/nvme0n1p3 -> Windows 11
/dev/nvme0n1p4 -> Newly created 20GB Btrfs

In archinstall just mounted EFI as /boot and btrfs as /.
2021.12.01 archiso
I forgot to take logfile but if necessary i can try again and take log.
![20211202_104351](https://user-images.githubusercontent.com/17825767/144379982-146579ea-be70-421a-a8f4-4bcbe51bc094.jpg)
![20211202_104330](https://user-images.githubusercontent.com/17825767/144379996-9353577b-3b55-4989-9bb2-a797120d68f6.jpg)

Edit: I saw something right now. In the second photo it shows PARTUUID for efi in "Re-using partition_instance" line but it shows as None in the next line.

**Comment by Torxed** [comment]
Thank you for posting this issue. I'll investigate and see what could cause this. But as you mentioned the device instance being re-used is probably the issue if the previous instance was `None` for some reason. I think it has to do with re-using partitions all together. So I think I know where to look, I'll come back to you if I really need the `install.log` in this instance.

**Comment by ndfr623** [comment]
Same here, except on a completely separate drive due to the Windows installer's bad habit of appropriating EFI partitions.
![IMG_20211206_114839731](https://user-images.githubusercontent.com/69735179/144845083-ddc86efd-4f78-43ab-a939-a4a2016edfa6.jpg)

**Comment by flechtwerkflechtwerk** [comment]
I get exactly the same error message when using archinstall. The partition selected for /boot is the Windows 10 EFI partition on one hard drive, the / partition is a newly formated ext4 partition on another drive. I did not create a log file but I can redo the install and create one if it would be helpful.

**Comment by wllacer** [comment]
This issue has happens un installer.ordered_mount*.
For some reason one partition element has no future mountpoint assigned, and It crashes all the process. I had a lot of them during my take on btrfs

**Comment by danboid** [comment]
I've just been hit by this bug trying to install arch alongside W10.

**Comment by TheDrawingCoder-Gamer** [comment]
same issue, trying to install arch alongside W10 as well

**Comment by phaitonican** [comment]
Same error

**Comment by paweljonik** [comment]
I had this error on NON-dual-boot disk with previously prepared partitions as follows:
1st vfat (not supported had to format to fat32) for /boot
2nd ext4 for /
3rd NTFS desired to be left alone.

And it turned out I couldn't leave 3rd partition as is - I had to at least indicate a mountpoint for it. Then the issue will not occur and the installer will proceed.

**Comment by Torxed** [comment]
I'll try to work on this one tomorrow. I'm pretty confident where the issue lies so I just need a moment to dig into the code :)

**Comment by RandomExplosion** [comment]
Same problem here when installing using pre-existing partitions:
0: vfat mounted at /boot/efi (hoped the installer wouldn't touch that as i had already set up secure boot on it)
1: ext4 mounted at /
2. ext4 mounted at /home

**Comment by Torxed** [comment]
Been tinkering with this issue for a bit now.
And it seems to only trigger when selecting a drive and using the guided manual partitioner.

The following works fine if you skip the partitioning step:
```
mkdir -p /mnt/archinstall
mount /dev/sdb3 /mnt/archinstall
mkdir -p /mnt/archinstall/boot
mount /dev/sdb1 /mnt/archinstall
archinstall
```

Which makes sense as there's no disk related steps involved in archinstall any longer.
But it's a work-around for now, but it won't "re-initate" the disk layout. So I'll continue to narrow down the issue and fix it : )

**Comment by Torxed** [comment]
I think the error relates to an issue with `lsblk` not able to detect and wait for `partprobe` to update cache, leaving some partitions with a `PARTUUID` of `None`. Which I fixed in 3ec9dbf a few minutes ago. I'll verify and if that works I'll pull in some changes from `master` which swaps out `lsblk` and uses `blkid` instead, which do detect pending cache updates *(or at least that's my laymen analysis of why `blkid` works and `lsblk` doesn't)*.

But for some reason I have a `"mountpoint": null` going on that I need to sort out as well after these fixes :)

![screenshot](https://user-images.githubusercontent.com/861439/147820259-9976eb02-0a14-428b-b680-23b98c1d027b.png)

**Comment by Torxed** [comment]
This is fixed, both in `master` and `v2.3.1-dev` and scheduled for release in v2.3.1 soon : ) Thank you for reporting and giving feedback on this issue.
